imap_processing/ialirt/tests/unit/test_decom_ialirt.py

```diff
@@ -0,0 +1,91 @@
+import sys
+from pathlib import Path
+
+import pytest
+
+from imap_processing.ialirt.l0.decom_ialirt import decom_packets, generate_xarray
+
+
+@pytest.fixture(scope="session")
+def packet_path():
+    """Returns the raw packet directory."""
+    return Path(sys.modules[__name__.split(
+        '.')[0]].__file__).parent / 'ialirt' / 'tests' / 'test_data' / 'l0' / \
+        'IALiRT Raw Packet Telemetry.txt'
+
+
+@pytest.fixture(scope="session")
+def xtce_ialirt_path():
+    """Returns the xtce auxiliary directory."""
+    return Path(sys.modules[__name__.split(
+        '.')[0]].__file__).parent / 'ialirt' / 'packet_definitions' \
+        / "ialirt.xml"
+
+
+@pytest.fixture(scope="session")
+def decom_data(packet_path, xtce_ialirt_path):
+    """Data for decom_ultra"""
+    data_packet_list = generate_xarray(packet_path, xtce_ialirt_path)
+    return data_packet_list
+
+
+@pytest.fixture(scope="session")
+def decom_test_data(packet_path, xtce_ialirt_path):
+    """Read test data from file"""
+    data_packet_list = decom_packets(packet_path, xtce_ialirt_path)
+    return data_packet_list
```

Comment:
These two seem really similar to me and I didn't quite know the different due to the variable namings both being list at first glance.

I _think_ if you move the `decom_packets` code you wrote into a test function here instead, then you could use `imap_processing.decom.decom_packets()` to do your packet list creation inside of `generate_xarray()` and not need both of these fixtures.